runelite-client/src/main/java/net/runelite/client/plugins/achievementdiary/Requirement.java

```diff
@@ -29,14 +29,22 @@
 @Getter
 public class Requirement
 {
-	private final Skill skill;
-	private final int levelRequirement;
-	private final Requirement[] altRequirements;
+	private Skill skill;
```

Comment:
Make these final and `null` the requirement type which isn't set. (skill or customRequirement, that is)